iceoryx_hoofs/test/moduletests/test_cxx_function.cpp

```diff
@@ -69,14 +69,22 @@ class Counter
         ++numDestroyed;
     }
 
-    Counter& operator=(const Counter&)
+    Counter& operator=(const Counter& rhs)
     {
-        ++numCopied;
+        if (this != &rhs)
```

Comment:
I am not sure it was initially meant to count the invocations of this Ctor or the actual copies. I think the latter. It should it not be used in self copy/move scenarios in the tests, so either is fine.